kedro/io/data_catalog.py

```diff
@@ -567,13 +638,42 @@ def list(self, regex_search: str | None = None) -> list[str]:
             ) from exc
         return [dset_name for dset_name in self._data_sets if pattern.search(dset_name)]
 
+    def exists_in_catalog(self, dataset_name: str) -> bool:
+        """Check if a dataset exists in the catalog as an exact match or if it matches a pattern."""
+        if (
+            dataset_name in self._data_sets
+            or dataset_name in self._pattern_name_matches_cache
+        ):
+            return True
+        matched_pattern = self.match_name_against_pattern(dataset_name)
+        if self.dataset_patterns and matched_pattern:
+            # cache the "dataset_name -> pattern" match
+            self._pattern_name_matches_cache[dataset_name] = matched_pattern
+            return True
+        return False
+
+    def match_name_against_pattern(self, dataset_name: str) -> str | None:
+        """Match a dataset name against existing patterns"""
+        # Loop through all dataset patterns and check if the given dataset name has a match.
+        for pattern in self._sorted_dataset_patterns:
+            result = parse(pattern, dataset_name)
+            # If there's a match resolve the rest of the pattern template to create
+            # a dataset instance. A result can be None or contain a dictionary of matched items:
```

Comment:
I think this comment is left over from the previous implementation and should be removed.